lib/types.ts

```diff
@@ -224,3 +224,9 @@ export type ReadonlyKeys<T extends object> = {
 export type WritableKeys<T extends {}> = {
   [P in keyof T]-?: IsFullyWritable<Pick<T, P>> extends true ? P : never;
 }[keyof T];
+
+/** Mark some properties which only the former including as optional and set the value to never */
+type Without<T, U> = { [P in Exclude<keyof T, keyof U>]?: never };
```

Comment:
The `Without` name of this helper type is, at least for me, not descriptive at all :(

Reply:
> The `Without` name of this helper type is, at least for me, not descriptive at all :(

so, what do u think? PickOverlapToNever? ... I think it‘s too redundant.

Reply:
If it's not _too_ long, it's better to have a kinda redundant name that helps to understand the type, than a short one that doesn't. But "overlap" is also not accurate, since you're _excluding_ the overlap it would rather be "pick excluded to never", but that starts to sound weird :D  You can also check out my reimplementation -- it may be that this helper type is too compilicated to begin with. But maybe I'm overreacting here -- @krzkaczor @macbem what are your opinions?